### ISSUE TITLE
Deprecated method: this method is going to be deleted. Use Fluent::MessagePackFactory.engine_factory

### ISSUE THREAD

**Issue by alex-vmw** [cmeerbeek/fluent-plugin-splunkhec#29]
When splunkhec plugin is loaded it produces a deprecation warning:
```
2021-10-26 22:28:44 +0000 [warn]: Deprecated method: this method is going to be deleted. Use Fluent::MessagePackFactory.engine_factory"
```
I don't really know the code, but I am guessing it is about https://github.com/cmeerbeek/fluent-plugin-splunkhec/blob/4ae4ea2b5242cb1deb1af3a1373afd26933e99f1/lib/fluent/plugin/out_splunkhec.rb#L48 and a possible fix would be to change that line as per below:
```
      @packer = Fluent::MessagePackFactory.engine_factory.packer
```

**Comment by cmeerbeek** [comment]
Thanks for raising this. I'll check it out and come back on this.

**Comment by cmeerbeek** [comment]
Fixed it and pushed v2.2 gem to rubygems.
